test/services/offer.service.test.js

```diff
@@ -492,6 +495,31 @@ describe('Offer Service', () => {
         expect(acceptedOffer.signature).to.eql(toAcceptInvalid.signature);
       });
     });
+
+    describe('when user was referred', () => {
+      const referral = ReferralFactory.build(
+        { referrerId: user2._id, userId: user1._id, 'reward.status': REWARD_STATUS.PENDING },
+        { generateId: true },
+      );
+
+      beforeEach(async () => {
+        await addReferral(referral);
+      });
+
+      context('when all is valid', () => {
```

Comment:
```suggestion
      context('when the referral is pending', () => {
```

We need to confirm that the reward status is only updated when it is `PENDING`. You might need to use a loop to test for other reward status.